admin/functional/test_aws.py

```diff
@@ -0,0 +1,53 @@
+import shutil
+import os
+from unittest import skipIf
+
+import boto
+from effect import Effect, sync_perform, ComposedDispatcher
+from twisted.python.filepath import FilePath
+from twisted.trial.unittest import SynchronousTestCase
+
+from ..aws import boto_dispatcher, UploadToS3
+from flocker.provision._effect import dispatcher as base_dispatcher
+from flocker.testtools import random_name
+
+# Bucket to use for testing
+bucket_name = 'clusterhq-archive-testing'
+
+try:
+    boto.connect_s3().head_bucket(bucket_name)
+    _can_connect = True
+except:
+    _can_connect = False
+
+if_aws = skipIf(not _can_connect, "Requires boto AWS credentials")
+
+
+class AWSTest(SynchronousTestCase):
+
+    @if_aws
+    def test_upload_content_type(self):
+        """
+        Test that a content type can be set for an uploaded file.
+        """
+        filename = random_name(self)
+        tmpdir = self.mktemp()
+        os.mkdir(tmpdir)
+        self.addCleanup(shutil.rmtree, tmpdir, ignore_errors=True)
```

Comment:
trial will clean up temporary directories from `self.mktemp` automatically on the next test run.  Better to let stuff like this stay around in case it ends up being useful for debugging.